pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,255 @@
+			expectOutput: "",
+		},
+		{
+			name: "valid report query with valid templating but PrestoTable contains empty schema field returns error and expected error output",
+			reportTemplate: &ReportQueryTemplateContext{
+				Namespace:         testNamespace,
+				Reports:           nil,
+				ReportQuery:       NewReportQueryCustom(ds1, "SELECT labels['namespace'] FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					NewPrestoTableCustom(ds1.Status.TableRef.Name, testNamespace, "", nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					ReportingStart: reportStart,
+					ReportingEnd:   reportEnd,
+					Inputs:         map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:35: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: prestoTable %s either contains an empty field in the Catalog, Schema, and/or TableName Status structure", ds1.Status.TableRef.Name),
+			expectOutput: "",
+		},
+	}
+
+	for _, testCase := range testTable {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			output, err := RenderQuery(testCase.reportTemplate, testCase.templateContext)
+
+			if testCase.expectErr {
+				assert.EqualErrorf(t, err, testCase.expectErrMsg, "expected that RenderQuery would return the correct error message")
+			} else {
+				assert.NoErrorf(t, err, "expected the report would return no error, but got an error.")
+				assert.EqualValuesf(t, testCase.expectOutput, output, "expected that RenderQuery would return the expected output")
+			}
+		})
+	}
+}
+
+// NewPrestoTableCustom allows us to insert a custom name, as opposed to testhelpers.NewPrestoTable
+func NewPrestoTableCustom(name, namespace, schema string, columns []presto.Column) *v1alpha1.PrestoTable {
+	return &v1alpha1.PrestoTable{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Status: v1alpha1.PrestoTableStatus{
+			Catalog:   "hive",
+			Schema:    schema,
+			TableName: name,
+			Columns:   columns,
+		},
+	}
+}
+
+// NewReportQueryCustomQuery helps create a ReportQuery with a custom ReportDataSource and adds a Query field
+func NewReportQueryCustom(dataSource *metering.ReportDataSource, query string) *metering.ReportQuery {
+	newDefault := func(s string) *json.RawMessage {
+		v := json.RawMessage(s)
+		return &v
+	}
+
+	return &metering.ReportQuery{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "unready-deployment-replicas",
+			Namespace: testNamespace,
+		},
+		Spec: metering.ReportQuerySpec{
+			Inputs: []metering.ReportQueryInputDefinition{
```

Comment:
This could probably just be passed in as an argument right?

Reply:
And you could have a function that creates an InputDefinition given a DataSource which you can use to create the input before passing it in.

Reply:
Updated to include `newTestInputDefinition`. may need to move `newDefault` into its own function or deflate it into the `Default` field